src/Internal/Plutip/Types.purs

```diff
@@ -69,10 +73,19 @@ type PlutipConfig =
   , customLogger :: Maybe (LogLevel -> Message -> Aff Unit)
   , suppressLogs :: Boolean
   , hooks :: Hooks
-  , clusterConfig ::
-      { slotLength :: Seconds }
+  , clusterConfig :: ClusterConfig
   }
 
+data ClusterConfig
+  = DefaultClusterConfig
+  | ClusterConfig
+      { slotLength :: Seconds
+      -- Adjust the max transaction size. Useful for debugging with traces
+      , maxTxSize :: UInt
+      -- Factor by which to increase the Max ex-units of a cluster
+      , increasedExUnits :: UInt
+      }
```

Comment:
I think ideally this would be just a `type` and remove the constructors, and instead provide a `defaultClusterConfig` variable with the default values, similar to how we've done things [here](https://github.com/Plutonomicon/cardano-transaction-lib//blob/cebf7f6aa93107b2756e3761fd0545d9d0939971/src/Internal/QueryM/ServerConfig.purs#L31). But my understanding is: the current behaviour is to defer choosing the default values to Plutip, and this change would make it harder to keep this behaviour?